org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/CodeActionFactory.java

```diff
@@ -117,4 +124,37 @@ public static CodeAction replaceAt(String title, String replaceText, TextDocumen
 		insertContentAction.setEdit(workspaceEdit);
 		return insertContentAction;
 	}
+
+	/**
+	 * Makes a CodeAction to create a file and add content to the file.
+	 * 
+	 * @param title The displayed name of the CodeAction
+	 * @param docURI The file to create
+	 * @param content The text to put into the newly created document.
+	 * @param diagnostic The diagnostic that this CodeAction will fix
+	 */
+	public static CodeAction createDocumentWith(String title, String docURI, String content, Diagnostic diagnostic) {
+
+		List<Either<TextDocumentEdit, ResourceOperation>> actionsToTake = new ArrayList<>(2);
+		
+		actionsToTake.add(Either.forRight(new CreateFile(docURI, new CreateFileOptions(false, true))));
+		
+		VersionedTextDocumentIdentifier identifier = new VersionedTextDocumentIdentifier(docURI, 0);
+		TextEdit te = new TextEdit(new Range(new Position(0, 0), new Position(0, 0)), content);
+		actionsToTake.add(Either.forLeft(new TextDocumentEdit(identifier, Collections.singletonList(te))));
+		
+		WorkspaceEdit createAndAddContentEdit = new WorkspaceEdit();
+		createAndAddContentEdit.setDocumentChanges(actionsToTake);
+
+		Command command = new Command();
+		command.setCommand(APPLY_WORKSPACE_EDIT);
+		command.setArguments(Collections.singletonList(createAndAddContentEdit));
+		
+		CodeAction codeAction = new CodeAction(title);
+		codeAction.setCommand(command);

```

Comment:
CreateFile is a LSP specification, it means that you should not have to create a custom command. 

Remove the custom cammand and replace this line with:

```java
codeAction.setEdit(createAndAddContentEdit);
```

I think it should work, because in TypeScript they do that:

https://github.com/microsoft/vscode-languageserver-node/blob/08e1782851e96be0382ae7c5fe204f23d4a9f957/testbed/server/src/server.ts#L400

Reply:
Thank you! Works perfectly.